topics/clusterwide-config.md

```diff
@@ -0,0 +1,208 @@
+Every instance in a cluster stores a copy of it.
+
+Clusterwide configuration contains options that must be identical on
+every cluster node, such as the topology of the cluster, failover and vshard
+configuration, authentication parameters and ACLs, and user-defined
+configuration.
+
+Clusterwide configuration doesn't provide instance-specific parameters:
+ports, workdirs, memory settings, etc.
+
+## Internal representation
+
+In the file system, clusterwide configuration is represented by a **file
+tree**. Inside `workdir` of any configured instance you can find the
+following directory:
+
+```text
+config/
+├── auth.yml
+├── topology.yml
+└── vshard_groups.yml
+```
+
+This is the clusterwide configuration with three default **config
+sections** - `auth`, `topology`, and `vshard_groups`.
+
+Due to historical reasons clusterwide configuration has two appearances:
+* old-style single-file `config.yml` with all sections combined, and
+* modern multi-file representation mentioned above.
+
+Before cartridge v2.0 it used to look as follows, and this representation is
+still used in HTTP API and luatest helpers.
+
+```yaml
+# config.yml
+---
+auth: {...}
+topology: {...}
+vshard_groups: {...}
+...
+```
+
+Beyond these essential sections, clusterwide configuration may be used
+for storing some other role-specific data. Clusterwide configuration
+supports YAML as well as plain text sections. It can also be organized
+in nested subdirectories.
+
+In Lua it's represented by the `ClusterwideConfig` object (a table with
+metamethods). Refer to the `cartridge.clusterwide-config` module
+documentation for more details.
+
+## Two-phase commit
+
+Cartridge manages clusterwide configuration to be identical everywhere
+using the two-phase commit algorithm implemented in the `cartridge.twophase`
+module. Changes in clusterwide configuration imply applying it on
+every instance in the cluster.
+
+Almost every change in cluster parameters triggers a two-phase commit:
+joining/expelling a server, editing replica set roles, managing users,
+setting failover and vshard configuration.
+
+Two-phase commit requires all instances to be alive and healthy,
+otherwise it returns an error.
+
+For more details, please, refer to the
+`cartridge.config_patch_clusterwide` API reference.
+
+## Managing role-specific data
+
+Beside system sections, clusterwide configuration may be used for storing
+some other **role-specific data**. It supports YAML as well as plain
+text sections. And it can also be organized in nested subdirectories.
+
+Role-specific sections are used by some third-party roles, i.e.
+[sharded-queue](https://github.com/tarantool/sharded-queue) and
+[cartridge-extensions](https://github.com/tarantool/cartridge-extensions).
+
+A user can influence clusterwide configuration in various ways. You can
+alter configuration using Lua, HTTP or GraphQL API. Also there are
+(luatest)[https://github.com/tarantool/luatest] helpers available.
```

Comment:
It looks as 
![image](https://user-images.githubusercontent.com/8830475/80472836-f25eaf00-894d-11ea-8cd9-4a54a4761b11.png)